projects/scripts/adi_board.tcl

```diff
@@ -933,18 +936,44 @@ proc ad_mem_hpx_interconnect {p_sel p_clk p_name} {
 ## Create an AXI4 Lite memory mapped interface connection for register maps,
 #  instantiates an interconnect and reconfigure it at every process call.
 #
+#  \param[p_sel] - name of the high speed interface
+#                  valid values are HPM0_FPD, HPM1_FPD, HPM0_LPD
 #  \param[p_address] - address offset of the IP register map
 #  \param[p_name] - name of the IP
 #  \param[p_intf_name] - name of the AXI MM Slave interface (optional)
 #
-proc ad_cpu_interconnect {p_address p_name {p_intf_name {}}} {
+proc ad_hpmx_interconnect {p_sel p_address p_name {p_intf_name {}}} {
 
   global sys_zynq
+  global sys_hpm0fpd_interconnect_index
+  global sys_hpm1fpd_interconnect_index
   global sys_cpu_interconnect_index
 
-  set i_str "M$sys_cpu_interconnect_index"
-  if {$sys_cpu_interconnect_index < 10} {
-    set i_str "M0$sys_cpu_interconnect_index"
+  if { $sys_zynq != 2 } {
+    puts "WARNING: ad_hpmx_interconnect: partial support for sys_zynq != 2"
+  }
+
+  if {($p_sel eq "HPM0_FPD") && ($sys_zynq == 2)} {
```

Comment:
If we use sys_cpu_interconnect_index for all, this part is unnecessary. except for setting set interconnect_name, but there must be a better way.
I would set interconnect_name axi_hpm*fpd_interconnect just by `p_sel eq "HPM*_*PD"` instead of `($p_sel eq "HPM*_*PD") && ($sys_zynq == 2)`, looks redundant and vivado sure will happily throw a critical warning if you use a interface that does no exist.
Why is HPM0_LPD called axi_cpu_interconnect instead of, for example, axi_hpm0lpd_interconnect ?